### PR TITLE
Registration service and it's tests was implemented

### PR DESCRIPTION
**Pull request by not4und**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -5,10 +5,30 @@
...
+        String login = user.getLogin();
+        if (login.isBlank()) {
+            throw new RuntimeException("User login shouldn't be null or empty.");
+        }
+        if (storageDao.get(login) != null) {
+            throw new RuntimeException("User already exists.");
+        }
+        Integer age = user.getAge();
+        if (age == null || age < MIN_USER_AGE) {
+            throw new RuntimeException("User's age shouldn't be null and less than 18.");
```
Comment (aaananas) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:26]:
```suggestion
            throw new RuntimeException("User's age shouldn't be null and less than " + MIN_USER_AGE);
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:19]:
```suggestion
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        user.setLogin("someLogin");
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:25]:
```suggestion
        user = new User();
        user.setLogin("someLogin");
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        user.setPassword("");
+        assertThrowRuntimeException();
+    }
+
+    @Test
+    void register_invalidPassword_notOk() {
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:66]: add test for short password

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        assertThrowRuntimeException();
+    }
+
+    @Test
+    void register_userCorrect_ok() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    private void assertThrowRuntimeException() {
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:89]: remove this method just write this line "asserThrows(.....)"each time when you need it

**Review by aaananas — CHANGES_REQUESTED** [review]

**Review by zachar-chorny — APPROVED** [review]
good job

**Pull request closed without merging**